File: backend/crates/akashic-extraction/tests/fixtures/kotlin/05-nested.kt
fun outer(items: List<Int>): Int {
    val factor = 2
    fun scale(x: Int): Int {
        return x * factor
    }
    fun total(): Int {
        return items.map { scale(it) }.sum()
    }
    return total()
}

fun standalone(): Int {
    return outer(listOf(1, 2, 3))
}

File: backend/crates/akashic-extraction/tests/fixtures/javascript/05-nested.js
export function outer(items) {
    const factor = 2;

    function scale(x) {
        return x * factor;
    }

    function total() {
        return items.map(scale).reduce((a, b) => a + b, 0);
    }

    const doubler = (y) => y * 2;
    return total();
}

export function standalone() {
    return outer([1, 2, 3]);
}
